docs/pages/machine-id/architecture.mdx

```diff
@@ -0,0 +1,154 @@
+---
+title: Machine ID Architecture
+description: How Teleport Machine ID works.
+---
+
+This section provides an overview of Teleport Machine ID's inner workings.
+
+A more in-depth specification of the workings of Machine ID can be found in
+[the Request For Discussion.](https://github.com/gravitational/teleport/blob/master/rfd/0064-bot-for-cert-renewals.md)
+
+## Bot creation
+
+When creating a bot via `tctl`, up to three resources are created:
```

Comment:
Which of these are optional/required? If it is up to three resources, what are the conditions where there would be less?